pkgs/desktops/deepin/deepin-screenshot/default.nix

```diff
@@ -0,0 +1,54 @@
+{ stdenv, fetchFromGitHub, cmake, pkgconfig, qttools, qtx11extras,
+  dtkcore, dtkwidget, dtkwm, deepin-turbo, deepin-shortcut-viewer,
+  deepin }:
+
+stdenv.mkDerivation rec {
+  name = "${pname}-${version}";
+  pname = "deepin-screenshot";
+  version = "4.1.8";
+
+  src = fetchFromGitHub {
+    owner = "linuxdeepin";
+    repo = pname;
+    rev = version;
+    sha256 = "1zhjchflgws2q74b9fgdjy2wzifjhimnrrq1sxlv395793xv8lrr";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    pkgconfig
+    qttools
+    deepin.setupHook
+  ];
+
+  buildInputs = [
+    # dde-file-manager
```

Comment:
did you already figure out whether `dde-file-manager` is needed or not?
Maybe add a small comment here…

Reply:
See https://github.com/NixOS/nixpkgs/pull/59242#issuecomment-481518554

Reply:
Right…

It might make sense it opens in a `deepin` desktop environment to open with `dde-file-manager`, but not in general. A hardcoded `/usr/bin/dde-file-manager` is ugly, for sure ;-)

I'm not sure how this is handled for example in gnome-shell, to make sure folders are opened with nautilus. Is that also just an `xdg-open` thing?

If that's the case, I'd be inclined to solely rely on `xdg-open` for deepin-screenshot, and even send the patch upstream (given it should work with xdg-open for them, too)

Reply:
> I'm not sure how this is handled for example in gnome-shell, to make sure folders are opened with nautilus. Is that also just an xdg-open thing?

It's just mime type associations. So the default file manager will be used, and in a deepin environment it would be `dde-file-manager`.

> If that's the case, I'd be inclined to solely rely on xdg-open for deepin-screenshot, and even send the patch upstream (given it should work with xdg-open for them, too) 

I'd say that's reasonable, though there's still room for improvement.

I'll put the patch here for now, and pr it to deepin :+1:

Reply:
* patch here https://github.com/NixOS/nixpkgs/pull/59242/commits/f3c5992644a0704408a193d517a51aa989cbea5e

Reply:
From the commit message:

```
deepin.deepin-screenshot: only use xdg-open

This doesn't hardcode or put it into PATH
as I wasn't sure what romildo would have preferred.
```

To have `xdg-open` in PATH is it enough to add it to the `buildInputs` list, or some sort of wrapping is also needed? If not needed, that is the easier choice, in my opinion.

Reply:
From the commit message:

```
deepin.deepin-screenshot: only use xdg-open

This doesn't hardcode or put it into PATH
as I wasn't sure what romildo would have preferred.
```

To have `xdg-open` in PATH is it enough to add it to the `buildInputs` list, or some sort of wrapping is also needed? If not needed, that is the easier choice, in my opinion.

Reply:
> To have xdg-open in PATH is it enough to add it to the buildInputs list, or some sort of wrapping is also needed? If not needed, that is the easier choice, in my opinion.

That's only in the build environment. Usually hardcoding, like you've been doing, is preferred.

You can also do
```
preFixup = ''
  gappsWrapperArgs+=(--prefix PATH : "${lib.makeBinPath [ xdg_utils ]}")
'';
```